jmix-dynattr/dynattr-flowui/src/main/java/io/jmix/dynattrflowui/view/categoryattr/CategoryAttributesDetailView.java

```diff
@@ -1137,4 +1163,32 @@ protected void preCommitConfiguration() {
             }
         }
     }
+
+    @Subscribe("isCollectionField")
+    public void onIsCollectionFieldComponentValueChange(final AbstractField.ComponentValueChangeEvent<JmixCheckbox, Boolean> event) {
+        setEnabledForDefaultValueControlsWithClearing(isScalar(), true);
+    }
+
+    @Subscribe("lookupField")
+    public void onLookupFieldComponentValueChange(final AbstractField.ComponentValueChangeEvent<JmixCheckbox, ?> event) {
+        setEnabledForDefaultValueControlsWithClearing(isScalar(), true);
+    }
+
+    protected boolean isScalar() {
+        return !isCollectionField.getValue();
+    }
+
+    protected void setEnabledForDefaultValueControlsWithClearing(boolean isEnabled, boolean clearValues) {
+        DEFAULT_VALUE_COMPONENT_IDS
+                .forEach((key, value) -> setEnabledAndClear(key, isEnabled && value.get(), clearValues));
+    }
+
+    protected void setEnabledAndClear(String componentId, boolean isEnabled, boolean clearValues) {
+        Component component = UiComponentUtils.getComponent(this, componentId);
+        ((HasEnabled)component).setEnabled(isEnabled);
```

Comment:
1. Better to check if a component implements an interface before cast (for both interfaces).
2. Format code